common/src/chain/transaction/signature/mod.rs

```diff
@@ -294,8 +297,15 @@ fn stream_signature_hash<T: Signable>(
     inputs.signature_hash(stream, mode, target_input, target_input_num)?;
     outputs.signature_hash(stream, mode, target_input, target_input_num)?;
 
-    // TODO: consider doing just like taproot, and hash in all outputs that come from the outpoints of inputs,
-    //       this would be a good solution to avoid having to download full transactions to verify inputs
+    // Include utxos of the inputs to make it possible to verify the inputs scripts and amounts without downloading the full transactions
+    if inputs.len() != inputs_utxos.len() {
+        return Err(TransactionSigError::InvalidUtxoCountVsInputs(
+            inputs_utxos.len(),
+            inputs.len(),
+        ));
+    } else {
```

Comment:
minor but else is unnecessary. Also, do we prefer `ensure!` for such situations?

Reply:
Done